core/server/data/migrations/versions/5.0/2022-04-25-10-32-backfill-mrr-for-discounted-subscriptions.js

```diff
@@ -0,0 +1,45 @@
+const logging = require('@tryghost/logging');
+
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        const subscriptionsToUpdate = await knex('members_stripe_customers_subscriptions AS s')
+            .join('offers AS o', 's.offer_id', '=', 'o.id')
+            .where('o.duration', '=', 'forever')
+            .andWhere('s.mrr', '!=', '0')
+            .select('s.*', 'o.discount_type AS offer_type', 'o.discount_amount AS offer_amount');
+
+        if (!subscriptionsToUpdate.length) {
+            logging.info('No subscriptions found needing updating');
+            return;
+        }
+
+        const toInsert = subscriptionsToUpdate.map((subscription) => {
+            let discountedAmount;
+            if (subscription.offer_type === 'percent') {
+                discountedAmount = subscription.plan_amount * (100 - subscription.offer_amount) / 100;
+            } else {
+                discountedAmount = subscription.plan_amount - subscription.offer_amount;
+            }
+
+            const newSubscription = {
+                ...subscription,
+                mrr: subscription.plan_interval === 'year' ? discountedAmount / 12 : discountedAmount
```

Comment:
Should we handle where `plan_interval` != `year`? I'm sure we've seen data where it's `week` before? 🤔

Reply:
I didn't bother because this is for offers, which are only for week and year